src/validation/mo_setsox/setsox_test.cpp

```diff
@@ -0,0 +1,342 @@
+// mam4xx: Copyright (c) 2022,
+// Battelle Memorial Institute and
+// National Technology & Engineering Solutions of Sandia, LLC (NTESS)
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <mam4xx/mam4.hpp>
+
+#include <mam4xx/aero_config.hpp>
+#include <skywalker.hpp>
+#include <validation.hpp>
+
+using namespace skywalker;
+using namespace mam4;
+using namespace haero;
+void setsox_test(Ensemble *ensemble) {
+  ensemble->process([=](const Input &input, Output &output) {
+    // Ensemble parameters
+    // Declare array of strings for input names
+    std::string input_variables[] = {"dt"};
+
+    std::string input_arrays[] = {
+        "ncol", "loffset", "dtime",  "press", "pdel", "tfld", "mbar",
+        "lwc",  "cldfrc",  "cldnum", "xhnm",  "qcw",  "qin"};
+
+    // Iterate over input_variables and error if not in input
+    for (std::string name : input_variables) {
+      if (!input.has(name.c_str())) {
+        std::cerr << "Required name for variable: " << name << std::endl;
+        exit(1);
+      }
+    }
+    // Iterate over input_arrays and error if not in input
+    for (std::string name : input_arrays) {
+      if (!input.has_array(name.c_str())) {
+        std::cerr << "Required name for array: " << name << std::endl;
+        exit(1);
+      }
+    }
+
+    // Parse input
+    const Real dt = input.get("dt");
+    const int ncol = input.get_array("ncol")[0];
+    const int loffset = input.get_array("loffset")[0];
+    Real dtime = input.get_array("dtime")[0];
+    Real press = input.get_array("press")[0];
+    Real pdel = input.get_array("pdel")[0];
+    Real tfld = input.get_array("tfld")[0];
+    Real mbar = input.get_array("mbar")[0];
+    Real lwc = input.get_array("lwc")[0];
+    Real cldfrc = input.get_array("cldfrc")[0];
+    Real cldnum = input.get_array("cldnum")[0];
+    Real xhnm = input.get_array("xhnm")[0];
+
+    auto qcw = input.get_array("qcw");
+    auto qin = input.get_array("qin");
+
+    output.set("qcw", qcw);
+    output.set("qin", qin);
+
+    mam4::mo_setsox::setsox();
+
+    // const Real zero = 0;
```

Comment:
Just an aside: While IDEs allow you to insert myriad single-line comments in blocks like the block below, but the block comment (`/* */`) syntax (or an `#ifdef 0`) is really a nice mechanism for this when working on a team with folks who use different editors.

Reply:
Ah, thank you! I'd never considered that, but will certainly adjust accordingly in the future